nautobot/core/ui/object_detail.py

```diff
@@ -1624,10 +1624,11 @@ def render_body_content(self, context: Context):
                     related_object_model_class, query = related_field
                 else:
                     related_object_model_class, query = related_field, f"{self.filter_name}__in"
-                filter_dict = {query: self.filter_pks}
-                related_object_count = (
-                    related_object_model_class.objects.restrict(request.user, "view").filter(**filter_dict).count()
-                )
+                filter_dict = {query: list(self.filter_pks)}
```

Comment:
Minor nit: line 1616 already constructs `self.filter_pks` as a list, I think the explicit cast is only needed at line 1612-1614. Moving the logic there would also avoid re-casting to a list on each iteration through the `self.related_models` loop, though I **think** that's harmless due to queryset caching?

Reply:
This change is what is introducing the extra query in the Location detail view, but as I understand it it is necessary at present in order to have the below `.distinct()` not raise an error, which is *probably* a bug in django-tree-queries but also *probably* not worth chasing down at this time.